src/redis-cache/redis-cache.service.ts

```diff
@@ -1,4 +1,14 @@
-import { Injectable } from '@nestjs/common';
+import { CACHE_MANAGER, Inject, Injectable } from '@nestjs/common';
+import { Cache } from 'cache-manager';
 
 @Injectable()
-export class RedisCacheService {}
+export class RedisCacheService {
+  constructor(@Inject(CACHE_MANAGER) private readonly cacheManager: Cache) {}
+
+  async clearCache(id?: string) {
```

Comment:
2 oddzielne metody to powinny byc. Masz tutaj 2 rozne konteksty

Reply:
w sensie nie mogę łączyć ze sobą operacji na redisie, które dotyczą różnych modułów lub metod?

Reply:
no powinienes zawsze tworzyc dedykowane rzeczy a nie flagowac metody ifem... Wzorzec Strategia o tym mowi chociazby ;)